code/modules/admin/verbs/playsound.dm

```diff
@@ -81,76 +81,83 @@
 		to_chat(src, span_boldwarning("Youtube-dl was not configured, action unavailable"), confidential = TRUE) //Check config.txt for the INVOKE_YOUTUBEDL value
 		return
 
-	var/web_sound_input = input("Enter content URL (supported sites only, leave blank to stop playing)", "Play Internet Sound via youtube-dl") as text|null
+	var/web_sound_input = tgui_input_text(usr, "Enter content URL (supported sites only, leave blank to stop playing)", "Play Internet Sound", null)
+
 	if(istext(web_sound_input))
 		var/web_sound_url = ""
 		var/stop_web_sounds = FALSE
 		var/list/music_extra_data = list()
 		if(length(web_sound_input))
-
 			web_sound_input = trim(web_sound_input)
 			if(findtext(web_sound_input, ":") && !findtext(web_sound_input, GLOB.is_http_protocol))
 				to_chat(src, span_boldwarning("Non-http(s) URIs are not allowed."), confidential = TRUE)
 				to_chat(src, span_warning("For youtube-dl shortcuts like ytsearch: please use the appropriate full url from the website."), confidential = TRUE)
 				return
-			var/shell_scrubbed_input = shell_url_scrub(web_sound_input)
+			var/shell_scrubbed_input = shell_url_scrub(replacetext(web_sound_input, "music.", ""))
```

Comment:
explain

Reply:
ive noticed some issues with links played from https://music.youtube.com/ where in some regions the music. links will not play properly, this just was me autocorrecting that as just simply removing music. still works and corrects it for some regions

Reply:
youtube-dl supports **1225 websites**.

This needs to be more exact.

filtering on `music.` is guaranteed to lead to false positives.

Reply:
here is a neat rule when working on anything related to youtube-dl, assume it will almost never be used to download youtube videos.

youtube-dl supports MANY sites and a youtube centric approach to working with it is gonna lead you down the wrong path.

Reply:
alright then, i removed that filter